File: csharp/Services/CreditCardService.cs

using Camunda.Training.CSharp.Exceptions;
namespace Camunda.Training.CSharp.Services;

public class CreditCardService
{
  public void ChargeAmount(string? cardNumber, string? cvc, string? expiryDate, double amount)
  {
    if (expiryDate?.Length == 5)
    {
      Console.Out.WriteLine("Credit card number: " + cardNumber + " CVC: " + cvc + " Expiry date: " + expiryDate +
          " Amount to charge: " + amount);
    }
    else
    {
      string errorMessage = "Invalid credit card expiry date: " + expiryDate;
      Console.Error.WriteLine(errorMessage);
      throw new InvalidCreditCardException(errorMessage);
    }
  }
}

File: python/services/credit_card_service.py
# credit_card_service.py

def charge_credit_card(card_number: str, cvc: str, expiry_date: str, amount: float) -> None:
    """
    Simulates charging a credit card for the given amount.
    """
    if len(expiry_date) != 5:
        raise ValueError("expiryDate must be exactly 5 characters long")

    print(f"Card {card_number}, with CVC {cvc} and expiry date {expiry_date} was charged for {amount}")
